### PR TITLE
Add new controller to handle list updates

### PR DESCRIPTION
**Pull request by phamann**
This adds a new route `/list/update` to handle updating of list data and adding content to lists. Sadly in cannot be truly RESTful via a `PUT /list`  as HTML forms do not allow for a PUT method so have settled with its on route. :sadpandaface:

The main reason for doing this was to handle empty submissions of add content to list form, but has the added benefit that doing the work to edit lists should now be trivial.

### CODE REVIEW

File: server/controllers/lists/create.js [hunk 1]
```diff
@@ -13,7 +13,7 @@ module.exports = (req, res, next) => {
 		req.body.contentUuid = sanitizeHtml(req.body.contentUuid);
 	}
 
-	if(!req.body.name) {
+	if(!req.body.name || req.body.name === '') {
```
Comment (keirog) [inline server/controllers/lists/create.js]: But `!!'' === false` anyway, no?

File: server/controllers/lists/update.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+const apiProxy = require('../../lib/api-proxy');
+const sanitizeHtml = require('../../lib/sanitize-html');
+
+module.exports = (req, res, next) => {
+	let contentUuid;
+	let listId;
+	req.body.name = sanitizeHtml(req.body.name);
+
+	if(!req.body.uuid) {
```
Comment (laurieboyes) [inline server/controllers/lists/update.js]: These core experience error redirects are stale and possibly a bit rubbish anyway, I think Keith might have been looking at this for another controller

File: server/controllers/lists/update.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+	apiProxy({
+		actorName: 'list',
+		actorId: listId,
+		type: 'list',
+		method: 'put',
+		body: req.body
+	}, res.locals.flags)
+	.then(() => {
+
+		console.log('Updated list');
```
Comment (laurieboyes) [inline server/controllers/lists/update.js]: accidental log?

File: views/partials/list-form.html [hunk 4]
```diff
@@ -6,16 +6,17 @@
...
 
 		<form class="myft-list-form" method="POST"
-			  action="/__myft/api/core/list/contained/content?method=post">
-			<input type="hidden" name="uuid" value="{{contentId}}"/>
-				<div class="o-forms-group">
-					<select name="subject" class="o-forms-select js-list-select">
-						<option>Choose a list</option>
+			  action="/__myft/list/update">
+				<input type="hidden" name="contentUuid" value="{{contentId}}"/>
+				<div class="o-forms-group {{#ifEquals error 'noListId'}}o-forms--error n-myft-ui__error--no-name{{/ifEquals}}">
```
Comment (laurieboyes) [inline views/partials/list-form.html]: I guess as it's currently the only error on the form it doesn't matter that it's got a weird name but it still bothers me a little bit. I'm really not sure why I put these error styles in myft-ui (https://github.com/Financial-Times/next-myft-ui/blob/master/main.scss#L176-L183). Maybe move them back into myft-page?

Comment (phamann) [inline views/partials/list-form.html]: They are still needed in `myft-ui` as it is loaded in non myFT pages such as the homepage and articles.

**Comment by phamann** [comment]
Dependent on: https://github.com/Financial-Times/next-myft-ui/pull/81

**Comment by laurieboyes** [comment]
:+1:

**Pull request merged**